index.js

```diff
@@ -317,16 +335,31 @@ browserslist.coverage = function (browsers, country) {
     }, 0);
 };
 
-// Return array of queries from config content
-browserslist.parseConfig = function (string) {
-    return string.toString()
-            .replace(/#[^\n]*/g, '')
-            .split(/\n/)
-            .map(function (i) {
-                return i.trim();
-            }).filter(function (i) {
-                return i !== '';
-            });
+// Return array of queries from config content by env section
+browserslist.parseConfig = function (string, env) {
```

Comment:
Strange that you pass `env` to `parseConfig`, because `env` doesn’t change a way how you ”parse”. I understand that you tried to keep old code, but it will be better to return a object from `parseConfig`.